client/app/utils/url.js

```diff
@@ -73,9 +75,18 @@ const currySearchParams = function currySearchParams(restrict_to_params) {
   };
 };
 
+const upsertSearchQueryParam = function upsertSearchQueryParam(location, param, value) {
+  const originalParams = parseSearchQueryParams(location);
+  const newParams = { ...originalParams, [param]: value };
```

Comment:
*value* is probably now mutable since it's not cloned with Object.assign or spread operator. Although, this returns  a string map, so it doesn't matter.